### ISSUE TITLE
Duplicate focus name will cause database problems

### ISSUE THREAD

**Issue by revoltinstyle** [jordsta95/hoi4-national-focus-maker#7]
I have been creating a focus tree, but most of the time, once the focus is added. I always get multiple "Duplicate focus name will cause database problems:" And then after the colon it says most of my focuses.

Heres an example in my error.txt : 

[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: awakentheindustries
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: dockyards1
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: expandtheprivatesector
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: investinmilitary
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: expandtheprivatesector
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: rigorousmilitaryexpansion
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: rigorousmilitaryexpansion
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: expansionofbusiness
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: revampourlabs
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: revampourlabs
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: stormjerusalem
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: organizethecommunals
[20:58:33][nationalfocus.cpp:55]: Duplicate focus name will cause database problems: organizethecommunals

I am extremely thankful for this tool you've made, but I keep continually getting this issue.
All help is appreciated.

**Comment by xDangerRanger** [comment]
Yeah that first appeared when "my" export focus error thingy was fixed by jordasta (feb 11: fix focus not exporting; adding focuses to tree instead of processing them)...........................I know it doesn't really help you revolt, but it might help jordasta finding the "problem"

In then meantime you can open the focus trees TXT files yourself (with the standard notepad ot notepad++ or whatever editor you like) and manually search for focus duplicates and deleting them. I am doing that aaaaand it works....but it's time consuming for trees with 20/30+ focuses....whatever.
Man I am writing way too much. I should stop now.

**Comment by jordsta95** [comment]
This is going to be a stupid question, but...

Do you have multiple focuses with the same names (in multiple focus trees?)

**Comment by xDangerRanger** [comment]
(I think that question is for revolt and me? Whatever, still gonna answere)
No

**Comment by revoltinstyle** [comment]
I know for my focus tree I personally have no overlapping names for focuses, or multiple names that are the same for multiple focuses.

With every new export of my focus tree, and me putting it into my mod files, I regularly have to delete duplicates, this is also the case, not just in the focus tree file, but in the .yml localisation file, where there is duplicate names and descriptions also. So I have to clear duplicates in that file as well.

**Comment by jordsta95** [comment]
What browser are you using?

I am unable to replicate this issue

**Comment by xDangerRanger** [comment]
Me: Chrome
Tipp for replicating the issue: Have a big focus tree (10 focuses) with diffrent rewards

**Comment by revoltinstyle** [comment]
I use google chrome as well. 
Is it possible for me to save to the server and send the password so you could have to ability to personally look at the focuses?
BTW- I have 25 focuses currently in the tree if that if worth note. (it is not done yet.)

**Comment by jordsta95** [comment]
That would be great, thanks, as then I can see what you're seeing

**Comment by revoltinstyle** [comment]
This is the server password ; xtflxlv96i

**Comment by jordsta95** [comment]
This issue should now be resolved the next time you export your focus tree.

When the PHP script which loops over each focus to format them correctly it will put the focus ID into an array, and before the next focus gets looked at, the aforementioned array will be checked to make sure the focus ID has not already been added to the focus tree.

I have no idea why multiple of the focuses were being added. When using your focus tree, I saw the same results you were seeing, however, when exporting ones I have created/imported from text files I couldn't replicate this issue. So I've added the above in as a sanity check for when the front-end decides to do something stupid like that.
